src/software/simulation/physics_simulator_ball.h

```diff
@@ -33,10 +33,18 @@ class PhysicsSimulatorBall : public SimulatorBall
      *
      * @param func The function to perform on the physics ball
      */
-    Point checkValidAndReturnPoint(
-        std::function<Point(const std::shared_ptr<PhysicsBall>)> func) const;
-    Vector checkValidAndReturnVector(
-        std::function<Vector(const std::shared_ptr<PhysicsBall>)> func) const;
+    template <class T>
```

Comment:
We should add a `@tparam` to the jdoc explaining what `T` is.

Reply:
added